app/src/main/java/com/kinandcarta/create/proxytoggle/feature/manager/view/ProxyManagerFragment.kt

```diff
@@ -0,0 +1,66 @@
+package com.kinandcarta.create.proxytoggle.feature.manager.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import com.kinandcarta.create.proxytoggle.databinding.FragmentProxyManagerBinding
+import com.kinandcarta.create.proxytoggle.feature.manager.viewmodel.ProxyManagerViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class ProxyManagerFragment : Fragment() {
+
+    companion object {
+        fun newInstance() =
+            ProxyManagerFragment()
+
+        const val FI_ADDRESS = "192.168.1.215"
+        const val FI_PORT = "8888"
+    }
+
+    private val binding by lazy { FragmentProxyManagerBinding.inflate(layoutInflater) }
+    private val viewModel: ProxyManagerViewModel by viewModels()
```

Comment:
what's this `viewModels()`? can't you just inject it?

Reply:
You can't inject a ViewModel and tie it to the Fragment's lifecycle o.O
This function is from Jetpack `androidx.fragment.app.viewModels` that creates the VM with the factory using the Fragment's lifecycle.
This combined with `@InjectViewModel` from Hilt makes the magic of not having to `@Inject` the VM

Reply:
You can if you set things up in Dagger. Might be too much if this is all you have in the project though

Reply:
What would be the benefit anyway? 

With this implementation you don't need to create custom factories or setup anything else. You have a way to obtain a ViewModel given by the framework and a way of injecting the dependencies in the ViewModel given by Hilt. Easy peasy!

Reply:
You wouldn't need to setup any custom factories. I don't know why things are done like that on projects here. You can just setup injection with Dagger...
I read the documentation for Hilt and played with it a bit awhile ago. If you can just use `viewModels()` to get a view model even when it has dependencies without setting up something then that's fine I guess

Reply:
Yeah, `viewModels()` is nicer. Even with Hilt they encourage you to use the same and just annotate the ViewModel constructor with `@ViewModelInject` or something similar.

Reply:
I have seen the annotations. It is convenient, but really does tie the code base to Hilt because of adding those annotations it spreads to the code base, instead of remaining in the dependency injection package/module you define. Which would be an issue if you were to choose to remove Hilt or if the make breaking changes to its api, something likely to happen while it is still in alpha. Still seems this is the best option we will have. @Inject is a bit less of an issue as it isn't specifically tied to Dagger